yt/data_objects/construction_data_containers.py

```diff
@@ -950,6 +950,28 @@ def _fill_particles(self, part):
         for p in part:
             self[p] = self._data_source[p]
 
+    def _check_sph_type(self, fi):
+        field = fi.name
```

Comment:
since `field[0]` ends up being used all the time in the following I'd favour giving explicit names to the tuple elements to make it clearer and idiomatic
```suggestion
        ftype, fname = fi.name
```

Reply:
Done